### DISCUSSION TITLE
Active forks

### DISCUSSION THREAD

**Discussion by Nachtzuster** [mcguirepr89/BirdNET-Pi#1130] [Show and tell]
Hi All,

I've been building on @mcguirepr89's most exelent work to futher update and improve BirdNET-Pi. Since these changes are unlikely to get merged here, I'd like to point to my fork here. Maybe someone will find it useful.
https://github.com/Nachtzuster/BirdNET-Pi

Changes include:
- Bookworm support
- Expirimental support for writing transient files to tmpfs
- Rework analysis to consolidate analysis/server/extraction. Should make analysis more robust and slightly more efficient, -especially on installations with a large number of recordings
- Bump tflite_runtime to 2.11.0, it is faster
- Rework daily_plot.py (chart_viewer) to run as a daemon to avoid the very expensive startup-
- Bump apprise version, so more notification type are possible
- Fix: Setting Excluded/Custom species from the UI, that have a ' now works
- Fix: Setting apprise notification body and apprise notification title that include 'special' characters like "{} now is possible. So you can send json now
- Fix: add missing laguages for new model
- Fix: Daily Chart was not including new detections due to caching
- Fix: changing advanced settings was not updating on a fresh install ed. CHANNELS, RECORDING_LENGHT, ... simplify

anyone aware of other active forks?

**Comment by pddpauw** [comment]
yes, I have an active fork here for Pi5:

- right verions of php in the CaddyFile (current version is 8.2)
- disable Apache (which gets installed)
- enable Caddy as systemd service
- change permission in /home/pi/ to make this work (chmod 755 to this folder) - note that this is not ideal. Preferably, we move all www data to /var/www/. It works, but can be optimized.
- change in the requirement.txt file to tflite_runtime-2.14.0-cp311-cp311-manylinux_2_34_armv7l.whl
- refresh the database in the script plotly_streamlit.py (st.cache_resource.clear())
- disable the terminal from the webpage Importantly, if you wish to enable this, then feel free to adapt in views.php in your homepage/ folder (after installation), and uncomment line 264, and comment line 265
- change table in homepage to 100 species (change # readings in daily_plot.py (line 46)) AND optimize layout for the table
- use V2.4 Model V2
- OPTIONAL: optimize for 4G environment (limit data usage): no automatic refresh of tables (in overview.php), not showing spectrogram (they get created, but not shown in the homepage)

https://github.com/pddpauw/BirdPi

**Reply by Nachtzuster** [reply]
Ah yes, I looked at the changes you did for RPi5 while poking around to get Bookworm going, so Thanks!

**Comment by blogdemoi** [comment]
> Hi All,
> 
> I've been building on @mcguirepr89's most exelent work to futher update and improve BirdNET-Pi. Since these changes are unlikely to get merged here, I'd like to point to my fork here. Maybe someone will find it useful. https://github.com/Nachtzuster/BirdNET-Pi
> 

Thank you. I am currently running a Birdnet-Pi on a Le Potato card. I see on your fork that it is possible to migrate. What does it mean exactly? That I can switch from the original Birdnet-Pi to your fork? Will the migration work for Le Potato card?

**Reply by Nachtzuster** [reply]
The idea is that if you are currently running a fully up-to-date original BirdNET-Pi, you can switch to my fork and keep your data.
Of course, having a backup before doing that is recommended ;-)

yes, it will work on a Le Potato, in fact, that is what I have running as my 'production' system.

**Reply by blogdemoi** [reply]
Ok, thanks, I will try in the coming weeks.

**Reply by Nachtzuster** [reply]
One question though: what OS are you using?
I'm running https://www.armbian.com/lepotato/

**Comment by Aixsponsa** [comment]
I had a similar question to the one above. Is it as easy as running the installation and then using the migration instructions?

Are there any known bugs specific to your fork?


Thanks for adding to this project!

**Reply by Nachtzuster** [reply]
For new installs, I would recommend running the installer from my fork.
`curl -s https://raw.githubusercontent.com/Nachtzuster/BirdNET-Pi/main/newinstaller.sh | bash`
That way you can use Bookworm, not the aging Bullseye.

**Reply by Svardsten53** [reply]
Nice work! I have done an installation and it looks good as long as you are running English as language. I have opened two issues in your fork.

As some of you know I have done a translation to Swedish and some adaptions on the original BirdNET-Pi installation and this also worked on yours as far as I can see.  

I have a suggestion for the daily plot. I've never liked the original look, so I got help with an adaptation that I've been running ever since. However, this one will not work on your new installation. I guess it requires some modification of calls to services. If you find it useful, feel free to use it. The counter for each species is selectable and also the number of rows displayed. A big advantage is that each row takes up the same amount of space in the graph, regardless of the number of rows. It saves space. I have attached the file as a text file.
It looks like this: https://birdnet.svardsten.se

[daily_plot.txt](https://github.com/mcguirepr89/BirdNET-Pi/files/14308821/daily_plot.txt)


Here is an updated daily_plot which looks nicer.

[daily_plot2.txt](https://github.com/mcguirepr89/BirdNET-Pi/files/14314620/daily_plot2.txt)

**Reply by Aixsponsa** [reply]
> For new installs, I would recommend running the installer from my fork.
> `curl -s https://raw.githubusercontent.com/Nachtzuster/BirdNET-Pi/main/newinstaller.sh | bash`
> That way you can use Bookworm, not the aging Bullseye.

Looks like it worked perfectly. I upgraded to bookworm and am receiving an apache page instead of the birdnet page. Did I miss something? Does apache need to be disabled?

**Reply by Aixsponsa** [reply]
> For new installs, I would recommend running the installer from my fork.
> `curl -s https://raw.githubusercontent.com/Nachtzuster/BirdNET-Pi/main/newinstaller.sh | bash`
> That way you can use Bookworm, not the aging Bullseye.

Looks like it worked perfectly. I upgraded to bookworm and am receiving an apache page instead of the birdnet page. Did I miss something? Does apache need to be disabled?

**Reply by pddpauw** [reply]
yes...  several things need to be done for bookworm (disabling apache, enabling caddy, changing a line in the caddy file, changing user permissions).

All of this is done in my repository, you can find this here:

https://github.com/pddpauw/BirdPi

good luck!

**Reply by Nachtzuster** [reply]
@Searchyourname I'm not sure I understand: are you saying you started with Bullseye and then upgraded that system to Bookworm?

**Reply by Aixsponsa** [reply]
Correct. I migrated to your repo. Everything updated/upgraded fine. I then went through the steps to upgrade my pi to bookworm. 

I am guessing at this point I should just do a fresh install.

**Reply by Nachtzuster** [reply]
I see. If you want Bookworm you should indeed start with a fresh install. I should have been clearer on that, sorry.

**Comment by Oripy** [comment]
I'm very interested!
How can I switch to your fork?
I made some modification to my install (I translated the interface strings to French and added the latest Species range model from BirdNET-Analyser).

**Reply by Nachtzuster** [reply]
Not sure about this one. I guess you would loose your changes, like you would when you update the original BirdNet-Pi

**Reply by Oripy** [reply]
Using some git commands I managed to put my changes on a branch, pull your repo and then merged my branch to yours.
It seems to have worked (my translation has been kept). 
Used the installation script (lost my data since the DB seems to have been erased, not really an issue as I'm still testing and it was not "real" data). Now it is running (doing analysis and stuff), but the result is always :

`12:25:53---[server][INFO] 0.0;3.0-('Human_Human', 0.0)

12:25:53---[server][INFO] 3.0;6.0-('Human_Human', 0.0)

12:25:53---[server][INFO] 6.0;9.0-('Human_Human', 0.0)

12:25:53---[server][INFO] 9.0;12.0-('Human_Human', 0.0)

12:25:53---[server][INFO] 12.0;15.0-('Human_Human', 0.0)`

Any idea why?

**Reply by Nachtzuster** [reply]
the installation script will do that, unfortunatly :-( I've run ito that myself
`./scripts/update_birdnet.sh -b new_branch` is your friend 

not sure about the log:
Are you saying you're getting `Human_Human` although it is most certainly getting bird sounds without human noises (maybe in the backround)?

**Reply by Oripy** [reply]
I'm getting no detection whatsoever (the best confidence is 0%, and so I think the list is just listing the first entries, that seems to be Humans).
It's been up for more that 2h and did not get any detection although at this time of day it should get at least 50 detection per hour.

**Reply by Nachtzuster** [reply]
weird - I wonder if it might have to do with the privacy setting. Could you try setting it to zero?
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/5863099/d33172b4-8dcc-4b3a-98e8-70161780b35d)

**Reply by Oripy** [reply]
That did the trick, I got my first detection (a Short-toed treecreeper)!
I wonder why the Privacy Threshold is broken...

**Reply by Nachtzuster** [reply]
Glad to hear that!
Could be a bug from the original BirdNET-Pi, still it is possible I broke it somehow.

**Reply by Oripy** [reply]
It was working fine until until I switched to your repo. However I made a mess with my way to update, so it might still be an issue only on my install. I will track it down.

**Reply by Oripy** [reply]
Hum, still a bit suspicious about the result I get. I have a lot of Tawny owl (Strix aluco) detections between 2p.m and 3p.m which is suspicious (highest confidence is 94%).
I know that I have one near me, so it might have decided to call during the day, but it is suspicious...
I will listen to the recording when I have the chance.

**Comment by tastyfishes** [comment]
Hey, I migrated to your fork as I saw you had marked that apprise notifications with {} are fixed. However when sending `{"Common_Name": "$comname", "Scientific_Name": "$sciname"}` through MQTT it comes through as `\{"Common_Name": "New Holland Honeyeater", "Scientific_Name": "Phylidonyris novaehollandiae"\}`

I checked the Apprise version and it is 1.7.1, does that seem right?

**Reply by Nachtzuster** [reply]
That Apprise version does looks ok.
Could you open an issue in my repo?
Is this an actual notification or one sent though the 'Send Test Notification' button?

**Reply by tastyfishes** [reply]
Thanks for your reply, I tested it further and the actual notification works correctly but the 'Send Test Notification' doesn't. I'll open an issue for if you want to fix this.

**Comment by Sparkybreno** [comment]
Thankyou @Nachtzuster   I’ve switched to your fork.  Is there a way to run an update from the command line?  My pi crashed last night and the interface was unusable.  I’ve done a clean install and shall see how I go

**Comment by harrymnel** [comment]
@Nachtzuster would it be possible to utilise the newest BirdNet Analyzer version V2.4 - V2, Jan 2024?

**Reply by Nachtzuster** [reply]
Yes, I had it in the backlog.
It is in there if you want to enable it:
![afbeelding](https://github.com/mcguirepr89/BirdNET-Pi/assets/5863099/b3260672-935f-43f4-8cae-9827a8fd73a0)
